src/main/java/com/cleanroommc/groovyscript/compat/mods/advancedrocketry/BaseMultiblockRegistry.java

```diff
@@ -0,0 +1,69 @@
+package com.cleanroommc.groovyscript.compat.mods.advancedrocketry;
+
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import net.minecraft.item.ItemStack;
+import zmaster587.libVulpes.interfaces.IRecipe;
+
+import java.util.*;
+
+public abstract class BaseMultiblockRegistry extends BaseRegistry {
+
+    @MethodDescription(example = @Example(priority = 2000, commented = true))
+    public void removeAll() {
+        List<IRecipe> recipes = getRecipeList();
+        recipes.forEach(this::addBackup);
+        recipes.clear();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.QUERY)
+    public SimpleObjectStream<IRecipe> streamRecipes() {
+        List<IRecipe> recipes = getRecipeList();
+        return new SimpleObjectStream<>(recipes)
+            .setRemover(this::removeRecipe);
+    }
+
+    public static abstract class RecipeBuilder extends BaseRegistry.RecipeBuilder {
+
+        // GrS cannot recognize these annotations on the builder class itself, so we have to redefine the fields

```

Comment:
to resolve this, modify the `Builder` class to add this new method and modify the `gatherFields` method to use it.
```java
    private static List<Property> getPropertyAnnotationsFromClassRecursive(Class<?> clazz) {
        List<Property> list = new ArrayList<>();
        Collections.addAll(list, clazz.getAnnotationsByType(Property.class));
        Class<?> superclass = clazz.getSuperclass();
        if (superclass != null) list.addAll(getPropertyAnnotationsFromClassRecursive(superclass));
        return list;
    }

//then modify the gatherFields method to use this new method

                            // Attached to the class or any parent classes, to create/override requirements set in the parent
                            getPropertyAnnotationsFromClassRecursive(builderClass).stream().filter(r -> r.property().equals(field.getName())),
```

Reply:
will this break like every mod compat

Reply:
it will, instead of only collecting the `Property` annotations on the class, also collect those annotations on all parent classes. i dont believe there are currently situations that would be affected by this.